usagov/vote-gov-nvrf-app
# Vote-451: Prompt to alert user on browser back button

## Jira ticket

[Vote-451](https://cm-jira.usa.gov/browse/VOTE-451)

## Description

This PR will add a prompt to alert the user that they will lose data if they hit the browser back/refresh button. 

Decision tree:

1. If a user makes it to the first page and has not filled out any data then they will be allowed to hit the browser back/refresh 
2. If a user adds any data and hits the browser back/refresh button they will be given a prompt to alert them that if they continue they will lose the data and given the option to exit that path

This will help provide a good user experience and help users not make it to the end and lose data

## Deployment and testing

### Pre-deploy

n/a

### Post-deploy

n/a

### QA/Test

1. Work your way to the first page of the form - Personal Information
2. Do Not fill out any fields - hit the refresh button - the expected result is that the user will be taken back to log in 
3. Work your way back to the first page of the form - Personal Information
4. Fill out a field or many and hit the refresh button - the expected result is that the user will be given a prompt that they will lose all their data

[Preview Link](https://federalist-aef5b597-8e18-44b6-aeba-3fc3f17cdac1.sites.pages.cloud.gov/preview/usagov/vote-gov-nvrf-app/feature/vote-451-backbutton-promt/)

## Checklist for the Developer




## Checklist for the Peer Reviewers
